### DISCUSSION TITLE
Recursively enumeration of input dir

### DISCUSSION THREAD

**Discussion by ilvpozzd** [hairyhenderson/gomplate#1326] [Q&A]
If I set inputDir like:

dir/
- child1/template.tpl
- child2/child1/template.tpl
template.tpl

will all templates be process?

And if i place to that dir NOT template file, will it be process and return original file?

**Comment by hairyhenderson** [comment]
Hi @ilpozzd, sorry for the slow response.

> will all templates be process?

Directories should be read recursively. I realize that's not clear from the documentation, so [I'll clarify that](https://github.com/hairyhenderson/gomplate/pull/1338).

> And if i place to that dir NOT template file, will it be process and return original file?

From the docs:

> _[...] In this case all files in input directory will be processed as templates [...]_

(from [`--input-dir` and `--output-dir`](https://docs.gomplate.ca/usage/#input-dir-and-output-dir))

So, yes, "all files" means "all files". There is no specific "template" file type that gomplate cares about (the `.tpl` extension is just a convention, and not something gomplate cares about). If you want to exclude specific files from being processed, use [`excludes`](https://docs.gomplate.ca/config/#excludes).
